### ISSUE TITLE
[Report] Unable to install despite having an internet connection.

### ISSUE THREAD

**Issue by ghost** [holoiso-eol/holoiso#435]
## Your Issue:
```
Whenever I run the installer I get: 
"SSL_INIT
No Internet connection available, exiting
Use iwctl util to connect WiFi
Press any key to exit..."
I have tried connecting to wifi through iwctl and even using ethernet but I still get the same thing. I do have internet since I am able to ping google, so I am very lost on what I can try and do.
```

## Your device's full specs
```
CPU: Intel Atom x7-Z8750
GPU: Intel Integrated Graphics
I am trying to install this on a GPD Win 1 which is a fool's errand, I am a fool though.
```

## Your OS version (`cat /etc/os-release | grep VARIANT_ID | cut -d '"' -f 2`):
```
HoloISO_Stable_GUI-20220719_2019-x86_64.iso
```

## Your installation type:
```
N/A, Unable to install.
```

## Have you updated your installation recently?
```
N/A
```

**Comment by nex86** [comment]
I have the same issue on GPD Win 3. 
using the latest iso that boots into a live desktop.
I can't connect to any of my networks neither with iwctl or with the KDE hotspot selector.

**Comment by ghost** [comment]
I managed to install by going into the /bin/ folder and editing the holoinstall script with "sudo nano /bin/holoinstall" and removing the internet check from it.

**Comment by nex86** [comment]
I think its helpful having an internet connection for updating the packages.
if you have an external usb network adapter...

**Comment by jrcharney** [comment]
> I managed to install by going into the /bin/ folder and editing the holoinstall script with "sudo nano /bin/holoinstall" and removing the internet check from it.

This is correct. Commenting out the `wget` command near the beginning of the script definitely fixes this. While you are at it, comment out that if-else block that depends out the output of that `wget` command.  Thankfully, it is only used once.  This script would have been a complete mess if `wget` was used more than once.
 
The issue has to do with `wget` which for some reason throws that `SSL_INIT` error.  Personally, usually I use `curl` when I check to see if I have a connection. Generally a `curl ipinfo.io` (show IP information) or `curl wttr.in` (shows the weather) are good places to test a response. (Note: both of those sites rely on IP geolocation, YMMV if you use a VPN.)

`curl` is the better alternative to wget.  This lien below will return the HTTP code. If it returns `200` ("OK") that means the connection is successful.

```
curl -s -o /dev/null -I -w "%{http_code}" "https://github.com"`
```

Once that is done, you should be ready to go.

Honestly, I would prefer that the installation script use the command line rather than GTK. For starters, there's all these deprecation notices, secondly, I had to use `CTRL+ALT+F2` to get into a command line to fix this problem and log in as root (it would help to note what the root password is.)  After that I went back to the Plasma GUI using `CTRL+ALT+F7`. Again, it would have been nice to answer the set up questions in a text interface, preferably at the beginning of the install rather than in the middle.

Secondly, I had fixed the `SSL_INIT` issue after fixing that conflict between Network Manager and iwd, which I have something to say in another issue.

**Comment by theVakhovskeIsTaken** [comment]
wget is a bit wonky i guess, to be fixed later.

Noted.

**Comment by theVakhovskeIsTaken** [comment]
wget is a bit wonky i guess, to be fixed later.

Noted.
